### PR TITLE
Remove mainstream references

### PR DESCRIPTION
**Pull request by dwhenry**
Remove all reference to mainstream index from rummager. Ensuring the existing migration code is kept to assist with future migrations of government and detailed indices.

Reference to mainstream have been replaced with either government or govuk depending on the context of the tests, this has resulted in some data which does not match production, but does allow the tests to run.

https://trello.com/c/LiapHutO/553-remove-reference-to-mainstream-index

### CODE REVIEW

File: config/query/boosting.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 base:
   format:
-    # Mainstream formats
+    # Govuk formats
```
Comment (MatMoore) [inline config/query/boosting.yml]: I think both of these comments are misleading, we should just get rid of them

File: doc/content_overview.md [hunk 2]
```diff
@@ -8,9 +8,9 @@ and the apps currently responsible for publishing them as of June 2017.
 For a broader view of the content that is available, see [Document types on GOV.UK](https://docs.publishing.service.gov.uk/document-types.html).
 
 ## Publisher
-Publishes "mainstream" content on GOV.UK - about 3000 pages which target common user needs. This is the kind of stuff you can reach from the homepage.
+Publishes "govuk" content on GOV.UK - about 3000 pages which target common user needs. This is the kind of stuff you can reach from the homepage.
```
Comment (MatMoore) [inline doc/content_overview.md]: this mainstream is actually mainstream

File: doc/content_overview.md [hunk 3]
```diff
@@ -22,17 +22,16 @@ Implemented in [search_payload_presenter.rb](https://github.com/alphagov/publish
  - 20 places
  - 8 help pages
 
-
 ## Travel-advice-publisher
 Publishes [foreign travel advice](https://www.gov.uk/foreign-travel-advice) on GOV.UK. There are 226 pages, but they have
 a format of "custom application" in Rummager.
 
-Implemented in [search_payload_presenter.rb](https://github.com/alphagov/travel-advice-publisher/blob/master/app/presenters/search_payload_presenter.rb).
+Published via the publishing API
```
Comment (MatMoore) [inline doc/content_overview.md]: We could probably just delete this page from the docs. I just wrote it so I could quickly compare what the apps were doing

File: lib/search_config.rb [hunk 4]
```diff
@@ -102,7 +101,7 @@ def searcher
 
   def combined_index_schema
     @combined_index_schema ||= CombinedIndexSchema.new(
-      content_index_names,
+      all_index_names,
```
Comment (MatMoore) [inline lib/search_config.rb]: Why this change?

Comment (dwhenry) [reply lib/search_config.rb]: this all worked fine previously as govuk used the same field definitions as mainstream, however now that the mainstream schema has been removed this breaks in fun ways - like filter terms are removed as they lived on the mainstream schema. :)

File: doc/content_overview.md (hunk 3 above)
Comment (dwhenry) [reply doc/content_overview.md]: done

File: doc/content_overview.md (hunk 2 above)
Comment (dwhenry) [reply doc/content_overview.md]: reverted - nope.. removed

File: config/query/boosting.yml (hunk 1 above)
Comment (dwhenry) [reply config/query/boosting.yml]: removed

File: lib/search_config.rb (hunk 4 above)
Comment (MatMoore) [reply lib/search_config.rb]: I was confused by the definition of `all_index_names`:

```
  def all_index_names
    # this is used to process data in the rake file when `all` is passed in as previous we skipped `govuk`
    # we can't update index_names at this stage as it is used in multiple spots including the index filtering
    content_index_names + auxiliary_index_names + [govuk_index_name]
  end
```

Is it ok to include `auxiliary_index_names` here and in the test helpers?

Comment (dwhenry) [reply lib/search_config.rb]: umm.. no that is a good point.

I thought that was content + govuk only.. let me correct that

File: spec/support/integration_spec_helper.rb [hunk 5]
```diff
@@ -205,12 +205,12 @@ def stub_message_payload(example_document, unpublishing: false)
 private
 
   def build_sample_documents_on_content_indices(documents_per_index:)
-    SearchConfig.instance.content_index_names.each do |index_name|
+    SearchConfig.instance.all_index_names.each do |index_name|
```
Comment (MatMoore) [inline spec/support/integration_spec_helper.rb]: this still says `all_index_names` - not sure if it matters

File: spec/unit/schema/combined_index_schema_spec.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@
 RSpec.describe CombinedIndexSchema do
   before do
     search_config = SearchConfig.new
-    @index_names = search_config.content_index_names
+    @index_names = search_config.all_index_names
```
Comment (MatMoore) [inline spec/unit/schema/combined_index_schema_spec.rb]: and here

**Review by MatMoore — APPROVED** [review]
LGTM other than a couple of test things.
